### PR TITLE
Module to merge ObjectPointClouds from vision services

### PR DESCRIPTION
**Pull request by bhaney**
## Summary 

 Adds a new camera component `erh:vmodutils:object-pc-merge` that takes one or more vision services
  configured to return object point clouds, calls `GetObjectPointClouds` on each, and serves the union as
  a single `PointCloud` via `NextPointCloud` (and a 2D projection via `Images`). Also runs an opt-in
  cleaning pipeline on the merged cloud — outlier removal → largest-cluster keep → radius crop — to drop
  the wide ground-plane halo and stray noise that upstream segmenters typically leave behind, while
  preserving a tight cluster around the actual object.

  ## What's in here

  **New files**
  - `touch/object_pc_merge_camera.go` — registers `erh:vmodutils:object-pc-merge`, wires up the
  vision-service deps, implements `NextPointCloud` / `Images` / `Properties`.
  - `touch/object_pc_merge_clean.go` — cleaning helpers: `applyOutlier` (RDK's
  `pointcloud.StatisticalOutlierFilter`), `keepLargestCluster` (grid + union-find largest connected
  component, replaces the slow `Cluster()` for this hot path), `cropToRadius` (centroid-based bbox), and
  the `cleanMerged` orchestrator. Also exposes `fillCleaningDefaults` so existing configs pick up sensible
   defaults.
  - `touch/object_pc_merge_clean_test.go` — synthetic-cloud tests for each stage in isolation, edge cases
  (empty / small / no-cluster fallback / disabled), full-pipeline integration test, and benchmarks proving
   the fast cluster path is ~140× faster than the generic `Cluster()` on realistic noisy inputs.

  **Updated**
  - `meta.json` + README entry for the new model.
  - `cmd/module/cmd.go` — module registration.

  ## Cleaning pipeline (opt-out per-stage)

  After merging the per-service inputs, points flow through:

  1. **Statistical outlier filter** (`pointcloud.StatisticalOutlierFilter`, `meanK=50, stdDev=2.0` by
  default) — drops scattered fly-aways.
  2. **Largest connected component** (cell size = `cluster_max_distance`, default 10 mm; min 5 pts/cell,
  min 50 pts/component) — keeps only the cup/bottle and discards the disconnected ground halo.
  3. **Radius crop around centroid** (default 250 mm) — final safety net against any remaining outliers
  attached to the dense object via thin chains.

  "disable_cleaning" attribute skips all three stages. Defaults are tuned conservatively so existing configs of this module pick up cleaning automatically without losing object detail; explicit negative values disable a stage.

### CODE REVIEW

File: touch/object_pc_merge_camera.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+		return nil, resource.ResponseMetadata{}, err
+	}
+	return []camera.NamedImage{ni}, resource.ResponseMetadata{CapturedAt: time.Now()}, nil
+}
+
+func (opc *ObjectPCMergeCamera) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	return nil, nil
+}
+
+func (opc *ObjectPCMergeCamera) NextPointCloud(ctx context.Context, extra map[string]interface{}) (pointcloud.PointCloud, error) {
```
Comment (erh) [inline touch/object_pc_merge_camera.go]: if this starts getting used at all - we're going to want this to do the compute per call, but have a thread that does it when needed.

File: touch/object_pc_merge_camera.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/spatialmath"
+
+	"github.com/erh/vmodutils"
+)
+
+var ObjectPCMergeModel = vmodutils.NamespaceFamily.WithModel("object-pc-merge")
```
Comment (erh) [inline touch/object_pc_merge_camera.go]: i'ca ll "merge-all-objects-pointclouds"

**Review by erh — CHANGES_REQUESTED** [review]

File: touch/object_pc_merge_camera.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	big := pointcloud.NewBasicPointCloud(totalSize)
+
+	for _, pc := range inputs {
+		err := pointcloud.ApplyOffset(pc, nil, big)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	cleaned, err := cleanMerged(big, opc.cfg)
```
Comment (erh) [inline touch/object_pc_merge_camera.go]: i don't get why this is here, feels like it should be elsewhere

File: touch/merge_all_objects_camera.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+func (opc *ObjectPCMergeCamera) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	return nil, nil
+}
+
+func (opc *ObjectPCMergeCamera) NextPointCloud(ctx context.Context, extra map[string]interface{}) (pointcloud.PointCloud, error) {
+	inputs := []pointcloud.PointCloud{}
+	totalSize := 0
+
+	for _, svc := range opc.services {
+		objects, err := svc.GetObjectPointClouds(ctx, "", extra)
```
Comment (erh) [inline touch/merge_all_objects_camera.go:112]: does this method always have to return things in world frame?

**Review by erh — CHANGES_REQUESTED** [review]

File: touch/merge_all_objects_camera.go (hunk 4 above)
Comment (bhaney) [reply touch/merge_all_objects_camera.go:112]: Following the precedent set by the existing `pc-merge` camera in this same package (touch/pc_merge_camera.go), which merges point clouds from multiple cameras via pointcloud.ApplyOffset(pc, nil, big) with no per-input frame transform. it assumes all inputs are already in a common (world) frame.

The situation here is analogous: each PointCloudObject returned by a vision service is tied to the underlying camera that produced it, so the frame semantics are inherited from that camera the same way pc-merge inherits them from its source cameras. If we ever want configurable frame handling, it should land in both modules together rather than diverging here.

File: touch/object_pc_merge_camera.go (hunk 2 above)
Comment (bhaney) [reply touch/object_pc_merge_camera.go]: done, renamed

File: touch/object_pc_merge_camera.go (hunk 3 above)
Comment (bhaney) [reply touch/object_pc_merge_camera.go]: Pulled the cleaning out into its own top-level pcclean package so all the helper functions live there now. The merge camera just imports it.

**Review by erh — APPROVED** [review]

**Comment by erh** [comment]
ready to mege?

**Comment by bhaney** [comment]
> ready to mege?

yes
